ban/utils.py

```diff
@@ -57,3 +58,15 @@ def parse_mask(source):
                 parent[field] = {}
             parent = parent[field]
     return dest
+
+
+def cached_property(func):
```

Comment:
[Werkzeug is using a class for that](https://github.com/pallets/werkzeug/blob/master/werkzeug/utils.py#L35-L75) in order to keep the associated name/doc/module easily but YMMV.